test/ext/javy/javy_test.rb

```diff
@@ -0,0 +1,172 @@
+require "test_helper"
+require_relative "../../../ext/javy/javy.rb"
+
+class JavyTest < Minitest::Test
+  include TestHelpers::FakeFS
+
+  def setup
+    super
+    FileUtils.mkdir_p(::Javy::ROOT)
+    File.write(File.join(::Javy::ROOT, "version"), "v0.1.0")
+  end
+
+  def test_install_existing_version_for_mac_os
+    stub_executable_download
+
+    install(PlatformHelper.macos_config)
+
+    assert File.file?(Javy::TARGET)
+    assert File.executable?(Javy::TARGET)
+  end
+
+  def test_install_existing_version_for_windows
+    stub_executable_download
+
+    expected_target = Javy::TARGET + ".exe"
+    install(PlatformHelper.windows_config)
+
+    assert File.file?(expected_target)
+    assert File.executable?(expected_target)
+  end
+
+  def test_install_existing_version_for_linux
+    stub_executable_download
+
+    install(PlatformHelper.linux_config)
+
+    assert File.file?(Javy::TARGET)
+    assert File.executable?(Javy::TARGET)
+  end
+
+  def test_install_raises_for_http_errors_during_asset_download
+    simulate_broken_asset_link
+
+    assert_raises(Javy::InstallationError) do
+      install(PlatformHelper.macos_config)
+    end
+  end
+
+  def test_install_raises_for_incorrect_binary
+    stub_executable_download
+
+    File.expects(:executable?).with(Javy::TARGET).returns(false)
+
+    error = assert_raises(Javy::InstallationError) do
+      install(PlatformHelper.macos_config)
+    end
+
+    assert_equal "Failed to install javy properly", error.message
+  end
+
+  def test_build_runs_javy_command_on_unix
+    stub_executable_download
+    install(PlatformHelper.macos_config)
+
+    source = "src/index.js"
+    dest = "build/index.wasm"
+
+    Javy.any_instance.expects(:system).with(Javy::TARGET, source, "-o", dest, anything)
```

Comment:
Note that this `anything` is required here because Ruby 2.6.6 calls system with an extra argument. Here is a log for details https://github.com/Shopify/shopify-cli/runs/4152528644?check_suite_focus=true